### PR TITLE
pyobi: improve code structure, add PyObiBool and add more tests

### PR DESCRIPTION
**Pull request by prin-r**
## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: obi/pyobi/tests/test_pyobi.py [hunk 1]
```diff
@@ -2,7 +2,48 @@
...
-def test_encode_signed_integer():
+def test_encode_decode_bool():
+    # encode
+    assert PyObiBool().encode(True) == bytes.fromhex("01")
+    assert PyObiBool().encode(False) == bytes.fromhex("00")
+
+    # decode
+    assert PyObiBool().decode(bytes.fromhex("01")) == (True, b"")
+    assert PyObiBool().decode(bytes.fromhex("00")) == (False, b"")
+
```
Comment (taobun) [inline obi/pyobi/tests/test_pyobi.py:13]: For each type test add decode test that has left bytes after decode. Like assert PyObiBool().decode(bytes.fromhex("00aa")) == (False, bytes.fromhex("aa"))

**Review by taobun — CHANGES_REQUESTED** [review]

File: obi/pyobi/tests/test_pyobi.py (hunk 1 above)
Comment (prin-r) [reply obi/pyobi/tests/test_pyobi.py:13]: done

**Review by taobun — APPROVED** [review]

File: obi/pyobi/pyobi/pyobi.py [hunk 2]
```diff
@@ -26,28 +27,52 @@ def decode(self, data):
...
-            int.from_bytes(data[: self.size_in_bytes], byteorder="big", signed=self.is_signed,),
+            int.from_bytes(data[: self.size_in_bytes], byteorder="big", signed=self.is_signed),
             data[self.size_in_bytes :],
         )
 
 
-class PyObiVector(PyObiSpec):
-    REGEX = re.compile(r"^\[.*\]$")
+class PyObiBool(PyObiSpec):
+    def __init__(self, spec=""):
```
Comment (sorawit) [inline obi/pyobi/pyobi/pyobi.py]: Let's use default value "bool"

File: obi/pyobi/pyobi/pyobi.py [hunk 3]
```diff
@@ -113,11 +142,13 @@ def decode(self, data):
 
 
 class PyObiBytes(PyObiSpec):
-    REGEX = re.compile(r"^bytes$")
-
-    def __init__(self, spec):
+    def __init__(self, spec=""):
```
Comment (sorawit) [inline obi/pyobi/pyobi/pyobi.py]: bytes

File: obi/pyobi/pyobi/pyobi.py [hunk 4]
```diff
@@ -99,11 +126,13 @@ def decode(self, data):
 
 
 class PyObiString(PyObiSpec):
-    REGEX = re.compile(r"^string$")
-
-    def __init__(self, spec):
+    def __init__(self, spec=""):
```
Comment (sorawit) [inline obi/pyobi/pyobi/pyobi.py]: string

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**